fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java

```diff
@@ -1442,6 +1452,64 @@ public static TWorkGroup prepareResourceGroup(ConnectContext connect, ResourceGr
         return resourceGroup;
     }
 
+    private List<Pair<Long, TNetworkAddress>> adaptiveChooseNodes(PlanFragment fragment, Map<Long, ComputeNode> candidates,
+                                                                  Set<TNetworkAddress> childUsedHosts) {
+        List<Pair<Long, TNetworkAddress>> childHosts = Lists.newArrayList();
+        for (Map.Entry<Long, ComputeNode> entry : candidates.entrySet()) {
+            TNetworkAddress address = entry.getValue().getAddress();
+            if (childUsedHosts.contains(address)) {
+                childHosts.add(Pair.create(entry.getKey(), address));
+            }
+        }
+
+        long maxOutputOfRightChild = fragment.getChildren().stream()
+                .sorted(Comparator.comparing(e -> e.getPlanRoot().getId().asInt())).skip(1)
+                .map(e -> e.getPlanRoot().getCardinality()).reduce(Long::max)
+                .orElse(fragment.getChild(0).getPlanRoot().getCardinality());
+        long outputOfMostLeftChild = fragment.getChild(0).getPlanRoot().getCardinality();
+
+
+        long baseNodeNums = Math.max(1, maxOutputOfRightChild / TINY_SCALE_ROWS_LIMIT / fragment.getPipelineDop());
+        double base = Math.max(Math.E, baseNodeNums);
+
+        long amplifyFactor = Math.round(Math.max(1,
+                Math.log(outputOfMostLeftChild / TINY_SCALE_ROWS_LIMIT / fragment.getPipelineDop()) / Math.log(base)));
```

Comment:
can reuse `Math.max(1, maxOutputOfRightChild / TINY_SCALE_ROWS_LIMIT / fragment.getPipelineDop())`

Reply:
`amplifyFactor` is is determined by most left child while base node is determined by the largest output rows in the rest of children.